content/investigating-lang-ext-semantics/index.md

```diff
@@ -0,0 +1,211 @@
+categories = ["Haskell Foundation"]
+tags = ["Community", "Stability"]
++++
+
+We analyzed the head.hackage patches to understand 
+why code breaks on new GHC releases.
+Surprisingly, most breakage wasn’t caused by 
+Template Haskell—it came from deeper semantic changes in language extensions.
+This post walks through the main categories of breakage, 
+why they happened, and what they tell us about long-term stability.
+If you care about a smoother upgrade path for Haskell users, 
+we invite you to participate in the Stability Working Group.
+
+The [Haskell Foundation Stability Working Group](https://blog.haskell.org/stability-working-group/) 
+is interested in understanding
+why breakage occurs.
+This is an extension of our initial [investigation](https://jappie.me/analyzing-haskell-stability.html).
+We've recently done further analysis on [head.hackage](https://ghc.gitlab.haskell.org/head.hackage/),
+and learned surprisingly enough that the root cause
+of a lot of breakage isn't Template Haskell,
+but seems to be from language extension semantics[^meaning].
+We're doing this investigation to better understand where the Haskell
+Foundation stability working group should focus its efforts.
+About a year ago I started on analyzing the causes
+for all `head.hackage` patches, but I got bogged down
+by the sheer amount of work.
+Trevis suggested focusing on the
+most important question,
+why do language extension semantics cause so much breakage?
+So instead of being bogged down by analyzing all
+the patches, I just looked at the 12 patches from language extension
+semantics.
+
+[^meaning]: The precise meaning of features enabled by language extensions. I guess parser changes also count.
+
+This gave us the following table:
+
+| Name                                | Cause                                        | Had warnings? |
+|-------------------------------------+----------------------------------------------+---------------|
+| Cabal-2.4.1.0.patch                 | simplified subsumption                       | no            |
+| Cabal-3.0.2.0.patch                 | simplified subsumption                       | no            |
+| Cabal-3.2.1.0.patch                 | simplified subsumption                       | no            |
+| data-r-tree-0.6.0.patch             | parser change (see 1)                        | no            |
+| drinkery-0.4.patch                  | simplified subsumption                       | no            |
+| ghc-lib-parser-9.8.1.20231121.patch | rename forall identifiers (2)                | yes           |
+| hgeometry-ipe-0.13.patch            | Instances moved due to splice enforcement    | no            |
+| singletons-3.0.2.patch              | add TypeAbstractions as a language extension | yes           |
+| singletons-base-3.1.1.patch         | add TypeAbstractions as a language extension | yes           |
+| vector-space-0.16.patch             | * is type (4)                                | yes           |
```

Comment:
This lone `*` makes me frown. There is a possibility it is throwing the whole Markdown renderer off.
<img width="1292" height="291" alt="image" src="https://github.com/user-attachments/assets/3af11654-0bd7-4db0-bc94-2eeb490bb0ff" />

@jappeace It should probably be escaped?

Reply:
@jappeace welp it looks like that wasn't it. :/ Do you have a Markdown renderer on your computer that you could check against?

Reply:
https://github.com/haskell/blog.haskell.org/pull/26/commits/e16b382c7174502ed0c6f13517522d617bd4d35b